src/main/java/ch/njol/skript/expressions/ExprEntitySound.java

```diff
@@ -0,0 +1,154 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.Material;
+import org.bukkit.Sound;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class ExprEntitySound extends SimpleExpression<String> {
+
+	static {
+		if (Skript.methodExists(LivingEntity.class, "getDeathSound")) {
+			Skript.registerExpression(ExprEntitySound.class, String.class, ExpressionType.COMBINED,
+				"[the] (damage|hurt) sound[s] of %livingentities%",
+				"%livingentities%'[s] (damage|hurt) sound[s]",
+
+				"[the] death sound[s] of %livingentities%",
+				"%livingentities%'[s] death sound[s]",
+
+				"[the] [high:(tall|high)|(low|normal)] fall [damage] sound[s] [from [[a] height [of]] %-number%] of %livingentities%",
+				"%livingentities%'[s] [high:(tall|high)|low:(low|normal)] fall [damage] sound[s] [from [[a] height [of]] %-number%]",
+
+				"[the] swim[ming] sound[s] of %livingentities%",
+				"%livingentities%'[s] swim[ming] sound[s]",
+
+				"[the] [fast:(fast|speedy)] splash sound[s] of %livingentities%",
+				"%livingentities%'[s] [fast:(fast|speedy)] splash sound[s]",
+
+				"[the] eat[ing] sound of %livingentities% [(with|using|[while] eating [a]) %-itemtype%]",
+				"%livingentities%'[s] eat[ing] sound",
+
+				"[the] drink[ing] sound of %livingentities% [(with|using|[while] drinking [a]) %-itemtype%]",
+				"%livingentities%'[s] drink[ing] sound");
+		}
+	}
+
+	private static final int DAMAGE = 0;
+	private static final int DEATH = 2;
+	private static final int FALL = 4;
+	private static final int SWIM = 6;
+	private static final int SPLASH = 8;
+	private static final int EAT_WITH_ITEM = 10;
+	private static final int EAT = 11;
+	private static final int DRINK_WITH_ITEM = 12;
+	private static final int DRINK = 13;
+
+	private int sound;
```

Comment:
Can this be renamed to signify that it's of the sound pattern?
```suggestion
	private int soundPattern;
```